central/detection/lifecycle/manager_impl.go

```diff
@@ -402,14 +402,8 @@ func (m *managerImpl) UpsertPolicy(policy *storage.Policy) error {
 	}
 
 	if policies.AppliesAtRunTime(policy) {
-		// Perform notifications and update DB.
-		modifiedDeployments, err := m.alertManager.AlertAndNotify(lifecycleMgrCtx, nil, alertmanager.WithPolicyID(policy.GetId()))
-		if err != nil {
-			return err
-		}
-		if modifiedDeployments.Cardinality() > 0 {
-			defer m.reprocessor.ReprocessRiskForDeployments(modifiedDeployments.AsSlice()...)
-		}
+		// Perform notifications and update DB, reprocess risk
+		go m.processAlertsAndRisk(policy)
```

Comment:
It's hard to understand if this causes some sort of race condition as we are now allowing Central to start and potentially receive more runtime alerts that would be processed at the same time?

Reply:
We are processing alerts per policy in parallel

Reply:
Queues would also increase our memory so :-), but they would ensure processing order. There are bigger architectural changes here - with queues because we would decide - one queue per policy? one queue overall. If we plan to go that route, let's tackle this JIRA with Central HA rearch.